### ISSUE TITLE
haproxy probably not re-generating config during start

### ISSUE THREAD

**Issue by mnaiman** [opnsense/plugins#2474]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

[X] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md

[X] I have searched the existing issues and I'm convinced that mine is new.

[X] The title contains the plugin to which this issue belongs

**Describe the bug**
haproxy config files are not regenrated after restore of opnsense from XML backup and reboot

*Tip*: to validate your setup was working with the previous version, use opnsense-revert (https://docs.opnsense.org/manual/opnsense_tools.html#opnsense-revert)

**To Reproduce**
1)i was changing backends in haproxy gui
2)i wanted to restore XML backup because of lot of changes to haproxy and wanted to start changes from scratch
3)XML backup restore of whole device - reboot
4)haproxy behaved same as before restore
5)had to click apply to save to file system config from restored backup

**Expected behavior**
After restore XML backup service will act as defined in XML backup

**Screenshots**
NA

**Relevant log files**
NA

**Additional context**
NA

**Environment**
OPNsense 21.1

**Comment by fraenki** [comment]
> had to click apply to save to file system config from restored backup

I think that is how it currently is supposed to work. The restore proess does not trigger the creation of the configuration, AFAIK.

**Comment by fraenki** [comment]
> XML backup restore of whole device - reboot

I may have to correct myself: on reboot the configuration is automatically regenerated. If not, you have to check `System: Log Files: Backend` and `System: Log Files: General` for error messages related to it.

**Comment by sorano** [comment]
On this note. Is there any way to restore only the HAProxy config from full XML backup?

**Comment by fraenki** [comment]
I can't answer this question, since I've never used the backup restore feature.

**Comment by mnaiman** [comment]
> > XML backup restore of whole device - reboot
> 
> I may have to correct myself: on reboot the configuration is automatically regenerated. If not, you have to check `System: Log Files: Backend` and `System: Log Files: General` for error messages related to it.

No, there is are no errors. There cant be, config is right, just not regenerated after Restore/Reboot process :/

**Comment by r4nc0r** [comment]
This is also a Problem if Opnsense High Availability is used:
We have two Firewalls with HA configured. If i: 
1.  change the config on the primary 
2.  apply the changes
3.  sync to the secondary
4.  test a failover

Then the HaProxy on the secondary Firewall is not in the same state, because i need to apply the changes manually
